plugins/datadog/app/models/datadog_monitor.rb

```diff
@@ -44,28 +46,44 @@ def initialize(id, response = nil)
     @response = response
   end
 
-  def state
-    response[:overall_state]
+  # @return [String] nil, "Alert", "OK", "NoData"
+  # TODO: do we want to ignore muted / silenced scopes ?
+  def state(deploy_groups)
+    if match_source.present? && deploy_groups.presence
+      return nil unless groups = response.dig(:state, :groups)
+      deployed = deploy_groups.map { |dg| "#{match_target}:#{match_value(dg)}" }
+      alerting = groups.keys.flat_map { |g| g.to_s.split(",") }
+      (deployed & alerting).any? ? "Alert" : "OK"
+    else
+      response[:overall_state]
+    end
   end
 
   def name
     response[:name] || 'api error'
   end
 
   def url
-    "https://#{SUBDOMAIN}.datadoghq.com/monitors/#{@id}"
+    "#{BASE_URL}/monitors/#{@id}"
   end
 
   def reload
     @response = nil
   end
 
-  def alert?
-    state == "Alert"
-  end
-
   private
 
+  def match_value(deploy_group)
```

Comment:
what if you wanted two groups to match (say cluster and node type)?

Reply:
make a new PR ;)

Reply:
better to implement it right the first place so you don't have an ugly migration

Reply:
or just cheat and serialise it into the column :p

Reply:
all we need at is pod / cluster matching ... maybe env ... so idk why there should be more here ... what usecase do you have in mind ?

Reply:
hmm, i was thinking more for shell based deploys like sfn_kubernetes where we need cluster+nodetype. do they use the same code? i guess it can always come later

Reply:
we can always do comma-separated entries or something like that :)
going a bit too broad might not be that bad either, if something bad happens in the cluster you are deploying to that could be related ...